File: tasks/2025/round2_pp/tournament-pp.py
#!/usr/bin/env python3
# @check-accepted: examples N2 Nsmall no-limits
import sys

N = int(input().strip())

P = list(map(int, input().strip().split()))

winner = 0
runnerup = 0

A = max(P[:N//2])
B = max(P[N//2:])
if A > B:
    winner = P.index(A)
    runnerup = P.index(B)
else:
    winner = P.index(B)
    runnerup = P.index(A)


print(winner, runnerup)

sys.stdout.close()

File: tasks/2025/round2_pp/tournament-pp.cpp
// @check-accepted: examples N2 Nsmall no-limits
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);
    int N;
    cin >> N;

    vector<int> P(N);
    for (int i = 0; i < N; ++i) cin >> P[i];

    int winner = 0, runnerup = 0;
    winner = max_element(P.begin(), P.begin() + N / 2) - P.begin();
    runnerup = max_element(P.begin() + N / 2, P.end()) - P.begin();
    if (P[winner] < P[runnerup]) swap(winner, runnerup);

    cout << winner << " " << runnerup << endl;

    return 0;
}
